### PR TITLE
To provide compatibility with cmake >= 2.8

### PR DESCRIPTION
**Pull request by dvrogozh**
Fixes: #32

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 2.8)
```
Comment (uartie) [inline CMakeLists.txt]: GNUInstallDirs requires cmake >= 2.8.5

Comment (dvrogozh) [reply CMakeLists.txt]: Ok, I will change to >=2.8.5 then. Any other hits?

**Comment by yakuizhao** [comment]
@dvrogozh , can it be compiled and work well if it is based on 2.8.5 cmake?

**Comment by dvrogozh** [comment]
I don't have 2.8.5, but I checked with 2.8.12 on centos. It works here. Let me see if I can lay my hands on cmake 2.8.5. Hold on, please.

**Comment by uartie** [comment]
Perhaps you could just enforce >= 2.8.12 since you've tested it and it may be hard to find a distro with anything older than that.  I know for a fact that currently supported Ubuntu and Fedora releases all have cmake packages >= 3.5.

**Comment by uartie** [comment]
If users encounter a distro with something older, then they could file an issue at that time.  But at least we won't be erroneously trying to use non-existent features if they have an older untested version.

**Comment by yakuizhao** [comment]
Agree with what Arite mentioned.  Most ubuntu and fedora releases already use the cmake package >=3.5. And as the driver is built/validated on the cmake >=3.5,  it is unknown whether it has some issues when using the older version cmake. 
It is recommended that the cmake can be upgraded.

**Comment by uartie** [comment]
@yakuizhao In other words, I think if @dvrogozh confirms it works on 2.8.12, then making that the minimum required is a low risk

**Comment by yakuizhao** [comment]
If it is confirmed that it can work on 2.8.12,  it is also ok that the 2.8.12 is added as the lowest requirement. 
But as there exists some other versions between 2.8.12 and 3.5, we are not sure whether it can work on these versions. In such case it is much safer to use the 3.5 as the lowest requirement.

**Comment by dvrogozh** [comment]
2.8.12 can work. I use exactly this version on CentOS 7.4. As for whether we can build with the middle versions, I think that's some overthinking. After all, did you verify that you can build with 3.7, 3.8. 3.9 or 3.10? Looks like cmake actually respects backward compatibility, so we should expect that we will be able to build with >=2.8.12.

I will bump requirement to tested 2.8.12.

**Comment by dvrogozh** [comment]
2.8.5 actually won't work. I tried. So, that was a nice catch. There are 2 reasons to that fyi:
1. add_compile_options which appeared in 2.8.12
2. DIRECTORY added as a component to get_filename_component in 2.8.11 (PATH is legacy backward supported name)

2) is easily fixable, 1) is not that trivial. So, 2.8.12 looks really as a best minimum candidate,

**Comment by oliver-sang** [comment]
the main reason for media to use 3.5 is that gmmlib is using 3.5. media needs to build gmmlib then link it for now. So if media supplies a lower version then users will get wrong idea that she/he could use that version but finally she/he will see build failure quite unexpectedly.
cmrtlib does not depend on gmmlib, so no need to do this force.
Several media tools are pretty simple and were built standalone, so no need to do this force, either.

cmake version requirement is only for build machines, I don't believe for projects you mentioned that each target machine need to clone git repo then build from source, right? So I don't think we need to limit the cmake version according to target machine. It's pretty easy to install cmake version on (several) build machine(s) as I think. Thoughts?

**Comment by dvrogozh** [comment]
gmmlib can be built with cmake 2.8.12, they don't need 3.5. And I have submitted a pull request for them as well, here it is: https://github.com/intel/gmmlib/pull/2. I think that gmmlib owners simply had this version when they developed.

Folks, I think everyone understand that any version of cmake can be installed from sources and/or build server can be used. But as a developer I prefer to build on the target when I develop or investigate a bug. Additional step to transfer the built target from one machine to another is error prone and time consuming. Thus, I strongly believe that provide convenient option to develop is precious.

**Comment by uartie** [comment]
LGTM

**Comment by dvrogozh** [comment]
outdated

**Pull request closed without merging**